crates/bevy_reflect/bevy_reflect_derive/src/type_uuid.rs

```diff
@@ -50,24 +42,73 @@ pub(crate) fn type_uuid_derive(input: proc_macro::TokenStream) -> proc_macro::To
 
     let uuid =
         uuid.expect("No `#[uuid = \"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx\"` attribute found.");
+    gen_impl_type_uuid(TypeUuidDef {
+        type_ident,
+        generics: ast.generics,
+        uuid,
+    })
+}
+
+/// Generates an implementation of `TypeUuid`. If there any generics, the `TYPE_UUID` will be a composite of the generic types' `TYPE_UUID`.
+pub(crate) fn gen_impl_type_uuid(def: TypeUuidDef) -> proc_macro::TokenStream {
+    let uuid = def.uuid;
+    let mut generics = def.generics;
+    let r#type = def.type_ident;
```

Comment:
Nit: Still not a fan of `r#` prefixes when we have conventions like `ty` that make it a lot cleaner to use. Not a blocker but it'd be nice if this was renamed.

Reply:
yeah forgot to change this, will do